tools/gfx/renderer-shared.cpp

```diff
@@ -821,6 +821,62 @@ void ShaderProgramBase::init(const IShaderProgram::Desc& inDesc)
     }
 }
 
+Result ShaderProgramBase::compileShaders()
+{
+    // For a fully specialized program, read and store its kernel code in `shaderProgram`.
+    auto compileShader = [&](slang::EntryPointReflection* entryPointInfo,
+                             slang::IComponentType* entryPointComponent,
+                             SlangInt entryPointIndex)
+    {
+        auto stage = entryPointInfo->getStage();
+        ComPtr<ISlangBlob> kernelCode;
+        ComPtr<ISlangBlob> diagnostics;
+        auto compileResult = entryPointComponent->getEntryPointCode(
+            entryPointIndex, 0, kernelCode.writeRef(), diagnostics.writeRef());
+        if (diagnostics)
+        {
+            getDebugCallback()->handleMessage(
+                compileResult == SLANG_OK ? DebugMessageType::Warning : DebugMessageType::Error,
+                DebugMessageSource::Slang,
+                (char*)diagnostics->getBufferPointer());
+        }
+        SLANG_RETURN_ON_FAIL(compileResult);
+        SLANG_RETURN_ON_FAIL(createShaderModule(entryPointInfo, kernelCode));
```

Comment:
When I saw `create` it made it seem as if kernelCode was the output which made this suspicious.  Same point as earlier - perhaps better as `addShaderModule`.